src/event.rs

```diff
@@ -0,0 +1,623 @@
+
+use crate::{
+	hex_utils, ChannelManager, Error, FilesystemPersister, LdkLiteChainAccess, LdkLiteConfig,
+	NetworkGraph, PaymentInfo, PaymentInfoStorage, PaymentStatus,
+};
+
+#[allow(unused_imports)]
+use crate::logger::{
+	log_error, log_given_level, log_info, log_internal, log_trace, log_warn, FilesystemLogger,
+	Logger,
+};
+
+use lightning::chain::chaininterface::{BroadcasterInterface, ConfirmationTarget, FeeEstimator};
+use lightning::chain::keysinterface::KeysManager;
+use lightning::ln::PaymentHash;
+use lightning::routing::gossip::NodeId;
+use lightning::util::events as ldk_events;
+use lightning::util::persist::KVStorePersister;
+use lightning::util::ser::{MaybeReadable, Readable, ReadableArgs, Writeable, Writer};
+
+use bitcoin::secp256k1::Secp256k1;
+use rand::{thread_rng, Rng};
+use std::collections::{hash_map, VecDeque};
+use std::sync::{Arc, Condvar, Mutex};
+use std::thread;
+use std::time::Duration;
+
+/// The event queue will be persisted under this key.
+pub(crate) const EVENTS_PERSISTENCE_KEY: &str = "events";
+
+/// An event emitted by [`LdkLite`] that should be handled by the user.
+///
+/// This wraps an event of a specific type and will more often than not wrap a LDK event accessible
+/// via the `inner` field.
+///
+/// [`LdkLite`]: [`crate::LdkLite`]
+#[derive(Debug, Clone)]
+pub enum LdkLiteEvent {
+	/// A payment we sent was successful.
+	PaymentSuccessful(PaymentSuccessfulEvent),
+	/// A payment we sent has failed.
+	PaymentFailed(PaymentFailedEvent),
+	/// A payment has been received.
+	PaymentReceived(PaymentReceivedEvent),
+	// TODO: Implement after a corresponding LDK event is added.
+	//ChannelOpened(ChannelOpenedEvent),
+	/// A channel has been closed.
+	ChannelClosed(ChannelClosedEvent),
+	// TODO: Implement on-chain events when better integrating with BDK wallet sync.
+	//OnChainPaymentSent(OnChainPaymentSentEvent),
+	//OnChainPaymentReceived(OnChainPaymentReceivedEvent),
+}
+
+trait EventType {
+	const TYPE: u8;
+}
+
+impl Readable for LdkLiteEvent {
+	fn read<R: lightning::io::Read>(
+		reader: &mut R,
+	) -> Result<Self, lightning::ln::msgs::DecodeError> {
+		match Readable::read(reader)? {
+			PaymentSuccessfulEvent::TYPE => Ok(Self::PaymentSuccessful(Readable::read(reader)?)),
+			PaymentFailedEvent::TYPE => Ok(Self::PaymentFailed(Readable::read(reader)?)),
+			PaymentReceivedEvent::TYPE => Ok(Self::PaymentReceived(Readable::read(reader)?)),
+			// ChannelOpenedEvent::TYPE => {
+			// TODO ChannelOpened
+			//}
+			ChannelClosedEvent::TYPE => Ok(Self::ChannelClosed(Readable::read(reader)?)),
+			//5u8 => {
+			// TODO OnChainPaymentSent
+			//}
+			//6u8 => {
+			// TODO OnChainPaymentReceived
+			//}
+			_ => Err(lightning::ln::msgs::DecodeError::InvalidValue),
+		}
+	}
+}
+
+impl Writeable for LdkLiteEvent {
+	fn write<W: Writer>(&self, writer: &mut W) -> Result<(), lightning::io::Error> {
+		match self {
+			Self::PaymentSuccessful(event) => {
+				event.write(writer)?;
+				Ok(())
```

Comment:
Can one line all these like:

```rust
    Self::PaymentSuccessful(event) => Ok(event.write(writer)?),
```

Reply:
Ah, this is `cargo fmt` land, I don't make the rules 😁